server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/suite/UniqueIdTestWatcher.java

```diff
@@ -0,0 +1,138 @@
+  public static final String COPYBOOK_EXT = ".cpy";
+
+  @Override
+  public void testSuccessful(ExtensionContext context) {
+    interceptor(context);
+  }
+
+  @Override
+  public void testFailed(ExtensionContext context, Throwable cause) {
+    LOG.info("{} test failed", context.getRequiredTestMethod().toString());
+    interceptor(context);
+  }
+
+  @Override
+  public void testAborted(ExtensionContext context, Throwable cause) {
+    LOG.info("{} test is aborted", context.getRequiredTestMethod().toString());
+    interceptor(context);
+  }
+
+  @Override
+  public void testDisabled(ExtensionContext context, Optional<String> reason) {
+    LOG.info(
+        "{} test is disabled due to {}",
+        context.getRequiredTestMethod().toString(),
+        reason.orElse("unknown reason"));
+  }
+
+  @SneakyThrows
+  private static void interceptor(ExtensionContext context) {
+    Optional<String> rootDirectory =
+        Optional.ofNullable(System.getProperty(ROOT_DIRECTORY_PROPERTY));
+    if (rootDirectory.isPresent()) {
+      String rootFolder = rootDirectory.get();
+      PreprocessedDocument document = getDocumentFromContext(context);
+
+      Path mainFolderPath = TestWatcherHelper.fetchTargetPath(context, rootFolder);
+      writeCobolDocument(document, mainFolderPath);
+      writeCopybooks(document.getCopybooks(), mainFolderPath);
+      if (Objects.nonNull(document.getTestData())) {
+        writeTestData(document.getTestData(), mainFolderPath);
+        createZeroDiagnosticIndicatorFile(document.getTestData(), mainFolderPath);
+      } else {
+        LOG.info("no test data for {}", context.getRequiredTestMethod().toString());
+        Files.createFile(mainFolderPath.resolve("NoTestData"));
+      }
+    }
+  }
+
+  private static PreprocessedDocument getDocumentFromContext(ExtensionContext context) {
+    return (PreprocessedDocument)
+        context
+            .getStore(ExtensionContext.Namespace.create(context.getRequiredTestMethod()))
+            .get("document");
+  }
+
+  private static void writeCobolDocument(PreprocessedDocument document, Path mainFolderPath)
+      throws IOException {
+    Path cobolDocPath = mainFolderPath.resolve(SOURCE_CBL);
+    writeToFile(cobolDocPath, document.getText());
+  }
+
+  private static void writeCopybooks(List<CobolText> copybooks, Path mainFolderPath) {
+    copybooks.forEach(
+        cobolText -> {
+          try {
+            Path filePath = mainFolderPath.resolve(cobolText.getFileName() + COPYBOOK_EXT);
+            writeToFile(filePath, cobolText.getFullText());
+            System.out.println("Created file: " + filePath);
+          } catch (IOException e) {
+            System.err.println("Error writing copybook: " + e.getMessage());
+          }
+        });
+  }
+
+  private static void writeTestData(TestData testData, Path mainFolderPath) throws IOException {
+    Path testDataPath = mainFolderPath.resolve("testData.txt");
+    Files.deleteIfExists(testDataPath);
```

Comment:
I don't think these conditional deletions should be here... The code overall does not handle removal of copybooks, or NoTestData file etc.

Reply:
updated